pkg/csi/service/wcp/controller_helper.go

```diff
@@ -440,3 +443,88 @@ func getOverlappingNodes(accessibleNodes []string, topologyRequirement *csi.Topo
 	}
 	return overlappingNodes, nil
 }
+
+// canFakeAttach checks if the volume is eligible to be fake attached and returns a bool value
+func canFakeAttach(ctx context.Context, volumeID string, volumeManager cnsvolume.Manager) bool {
+	log := logger.GetLogger(ctx)
+	//Check volume annotations
+	pvcAnn, err := commonco.ContainerOrchestratorUtility.GetVolumeAnnotations(ctx, volumeID)
+	if err != nil {
+		log.Errorf("failed to get volume annotations")
+		return false
+	}
+
+	if val, found := pvcAnn[common.AnnIgnoreInaccessiblePV]; found && val == "yes" {
+		log.Infof("Found %s annotation for volume %s set to yes", common.AnnIgnoreInaccessiblePV, volumeID)
+		//Check if volume is inaccessible
+		log.Debugf("Checking volume health on CNS volume: %s", volumeID)
+		vol, err := common.QueryVolumeByID(ctx, volumeManager, volumeID)
+		if err != nil {
+			log.Errorf("failed to query CNS volume")
+			return false
```

Comment:
Suggestion: return error so that ControllerPublishVolume throws the error which gets bubbled up in Pod events.

Reply:
Done